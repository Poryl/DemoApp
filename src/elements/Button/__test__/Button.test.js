import React from 'react';
import {shallow} from 'enzyme';
import {CustomButton} from '../index';

describe('Button', () => {
  it('should render Button', () => {
    const wrapper = shallow(
      <CustomButton
        text="TEST"
        textSize={16}
        textColor="white"
        onPressEvent={() => {}}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render Button with disable state', () => {
    const wrapper = shallow(
      <CustomButton
        text="TEST"
        textSize={16}
        textColor="white"
        onPressEvent={() => {}}
        disabled
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render Button with button press', () => {
    const onPress = jest.fn();
    const wrapper = shallow(
      <CustomButton
        text="TEST"
        textSize={16}
        textColor="white"
        onPressEvent={onPress}
      />,
    );
    wrapper.simulate('press');
    expect(onPress).toHaveBeenCalled();
  });
});
