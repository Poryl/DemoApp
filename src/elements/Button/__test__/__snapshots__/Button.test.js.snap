// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render Button 1`] = `
<ForwardRef
  activeOpacity={0.5}
  onPress={[Function]}
>
  <View
    style={
      Object {
        "backgroundColor": "#0178FF",
        "borderRadius": 50,
        "padding": 10,
        "paddingHorizontal": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 16,
        }
      }
    >
      TEST
    </Text>
  </View>
</ForwardRef>
`;

exports[`Button should render Button with disable state 1`] = `
<ForwardRef
  activeOpacity={0.5}
  disabled={true}
  onPress={[Function]}
>
  <View
    style={
      Object {
        "backgroundColor": "#EDEDED",
        "borderRadius": 50,
        "padding": 10,
        "paddingHorizontal": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 16,
        }
      }
    >
      TEST
    </Text>
  </View>
</ForwardRef>
`;
